addons/pos_self_order/static/src/app/components/loading_overlay/loading_overlay.js

```diff
@@ -1,10 +1,12 @@
 import { Component, onMounted, useState } from "@odoo/owl";
+import { useSelfOrder } from "@pos_self_order/app/self_order_service";
 
 export class LoadingOverlay extends Component {
     static template = "pos_self_order.LoadingOverlay";
     static props = {};
 
     setup() {
+        this.selfOrder = useSelfOrder();
```

Comment:
what is the advantage of this change ?

Reply:
i saw that. That's the point. What's the advantage of doing that here instead of in the root component?

Reply:
the page is rendered twice, the onWillStart and other hooks will be called twice.